### PR TITLE
Cleaning Up Code

### PR DESCRIPTION
**Pull request by jeremylt**
This is a series of minor changes to clean up the codebase.

### CODE REVIEW

File: backends/ref/ceed-ref-basis.c [hunk 1]
```diff
@@ -84,17 +80,12 @@ static int CeedBasisApply_Ref(CeedBasis basis, CeedInt nelem,
...
-  }
-  if (emode & CEED_EVAL_GRAD) {
+  } break;
+  case CEED_EVAL_GRAD: {
     CeedInt P = basis->P1d, Q = basis->Q1d;
     // In CEED_NOTRANSPOSE mode:
-    // u is (P^dim x nc), column-major layout (nc = ncomp)
-    // v is (Q^dim x nc x dim), column-major layout (nc = ncomp)
+    // u is [dim, ncomp, P^dim, nelem], column-major layout
+    // v is [dim, ncomp, Q^dim, nelem], column-major layout
```
Comment (jedbrown) [inline backends/ref/ceed-ref-basis.c]: Same.

File: backends/optimized/ceed-opt-basis.c [hunk 2]
```diff
@@ -82,6 +84,7 @@ static int CeedBasisApply_Opt(CeedBasis basis, CeedInt nelem,
       post *= Q;
     }
   } break;
+  // Evaluate the gradient to/from quadrature points
```
Comment (jedbrown) [inline backends/optimized/ceed-opt-basis.c]: It's worth asking whether comments like this are providing new information or just rephrasing the code. Comments get stale easily so it's better if the code is clear with fewer comments.

**Comment by jeremylt** [comment]
This only touches the `/cpu/self` backends and the tests. I want to wait on renaming the variables in the frontend until after we finish the backend updates to not make life unnecessarily harder on other developers.

File: backends/optimized/ceed-opt-basis.c (hunk 2 above)
Comment (jeremylt) [reply backends/optimized/ceed-opt-basis.c]: Good point. Maybe leaving it in the Ref backend as that's the 'educational' backend and clearing it out of the Opt backend?

File: backends/template/ceed-tmpl.c [hunk 3]
```diff
@@ -36,6 +36,8 @@ static int CeedInit_Tmpl(const char *resource, Ceed ceed) {
   Ceed_Tmpl *impl;
   Ceed ceedref;
 
+  // Create refrence implimentation that implimentation will be dispatched
```
Comment (jedbrown) [inline backends/template/ceed-tmpl.c]: Check spelling and grammar.

File: backends/ref/ceed-ref-restriction.c [hunk 4]
```diff
@@ -31,17 +31,20 @@ static int CeedElemRestrictionApply_Ref(CeedElemRestriction r,
...
       for (CeedInt e = 0; e < nblk*blksize; e+=blksize)
         for (CeedInt j = 0; j < blksize; j++)
           for (CeedInt k = 0; k < ncomp*elemsize; k++)
             vv[e*elemsize*ncomp + k*blksize + j]
               = uu[CeedIntMin(e+j,r->nelem-1)*ncomp*elemsize + k];
     } else {
-      // vv is (elemsize x ncomp x nelem), column-major
-      // uu is (ndof x ncomp)
+      // Indicies provided, standard or blocked restriction
+      // vv is [elemsize, ncomp, nelem], column-major
```
Comment (jedbrown) [inline backends/ref/ceed-ref-restriction.c]: Not column major.

File: backends/optimized/ceed-opt-basis.c (hunk 2 above)
Comment (jedbrown) [reply backends/optimized/ceed-opt-basis.c]: Sounds reasonable.

**Comment by jeremylt** [comment]
I cleaned up the Doxygen output. I think it is much more clean and readable now.

**Comment by jeremylt** [comment]
OK @jedbrown, I'll stop fiddling with this branch so we can decide if we want to change any of it before merging. This has zero functionality changes and is purely cleanup and documentation changes.

File: include/ceed.h [hunk 5]
```diff
@@ -293,4 +303,16 @@ static inline CeedInt CeedPowInt(CeedInt base, CeedInt power) {
...
+  @brief Return mimimum of two integers
+
+  @param[in] a  The first integer to compare
+  @param[in] b  The second integer to compare
+
+  @return The minimum of the two integers
+
+  @ref Utility
+**/
+static inline CeedInt CeedIntMin(CeedInt a, CeedInt b) { return a < b ? a : b; }
```
[the remark this answers is not in the document]
Comment (jeremylt) [reply include/ceed.h]: Fixed up. I went with `CeedIntPow` but in retrospect `CeedPowInt` would have been less work.

**Comment by jedbrown** [comment]
@jeremylt I'm ready to merge if you approve of the changes I just pushed.

**Pull request merged**